File: python/large-class/good/payment_service.py
import secrets

class PaymentService:
    def __init__(self, payment_config):
        self.stripe_secret_key = payment_config.get('stripe_secret')
        self.paypal_client_id = payment_config.get('paypal_client_id')
        self.paypal_secret = payment_config.get('paypal_secret')

    def process_stripe_payment(self, amount: float, token: str):
        if token and amount > 0:
            return {'success': True, 'transaction_id': f'stripe_{secrets.token_hex(8)}'}
        return {'success': False, 'error': 'Invalid payment data'}

    def process_paypal_payment(self, amount: float, paypal_token: str):
        if paypal_token and amount > 0:
            return {'success': True, 'transaction_id': f'paypal_{secrets.token_hex(8)}'}
        return {'success': False, 'error': 'Invalid payment data'}

    def refund_payment(self, transaction_id: str, amount: float):
        if transaction_id.startswith('stripe_'):
            return {'success': True, 'refund_id': f'refund_{secrets.token_hex(8)}'}
        elif transaction_id.startswith('paypal_'):
            return {'success': True, 'refund_id': f'refund_{secrets.token_hex(8)}'}
        return {'success': False, 'error': 'Unknown transaction type'}

File: php/large-class/good/PaymentService.php
<?php

class PaymentService {
    private string $stripeSecretKey;
    private string $paypalClientId;
    private string $paypalSecret;

    public function __construct(array $paymentConfig) {
        $this->stripeSecretKey = $paymentConfig['stripe_secret'];
        $this->paypalClientId = $paymentConfig['paypal_client_id'];
        $this->paypalSecret = $paymentConfig['paypal_secret'];
    }

    public function processStripePayment(float $amount, string $token): array {
        // Simulate Stripe payment processing
        if ($token && $amount > 0) {
            return ['success' => true, 'transaction_id' => 'stripe_' . uniqid()];
        }
        return ['success' => false, 'error' => 'Invalid payment data'];
    }

    public function processPayPalPayment(float $amount, string $paypalToken): array {
        // Simulate PayPal payment processing
        if ($paypalToken && $amount > 0) {
            return ['success' => true, 'transaction_id' => 'paypal_' . uniqid()];
        }
        return ['success' => false, 'error' => 'Invalid payment data'];
    }

    public function refundPayment(string $transactionId, float $amount): array {
        if (strpos($transactionId, 'stripe_') === 0) {
            return ['success' => true, 'refund_id' => 'refund_' . uniqid()];
        } elseif (strpos($transactionId, 'paypal_') === 0) {
            return ['success' => true, 'refund_id' => 'refund_' . uniqid()];
        }
        return ['success' => false, 'error' => 'Unknown transaction type'];
    }
}
